### DISCUSSION TITLE
Welcome to caMicroscope Discussions!

### DISCUSSION THREAD

**Discussion by birm** [camicroscope/caMicroscope#659] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by PASEDA-MODUPE** [comment]
Hello everyone I'm Modupe Paseda, I got into Outreachy application phase for may 2024.super excited to participate contribute to caMicroscope community.

**Reply by We13b-MD** [reply]
Hello everyone I am Idundun Michael a fullstack developer (mern) I am very happy to be here.

**Comment by kivuvarosekivuvan** [comment]
Hello @birm👋,
My name is Rose Kivuva from Kenya (outreachy applicant 2024)and I am excited to join this community, eager to learn and contribute

**Reply by We13b-MD** [reply]
Nice meeting you Rose Kivuva

**Comment by ssemanich** [comment]
Hello am Semango Nicholas from kampala Uganda .Am an outreach intern looking forward to contributing to caMicroscope

**Comment by Chy-baby** [comment]
Hi everyone, My name is Chidera Ikpeama.  Outreachy applicant . So happy to be part of this contribution.

**Reply by gbengaoluwadahunsi** [reply]
Hello @Chy-baby, you are welcome here, I hope the following snippet guide you around the channels,

Kindly follow the following steps

There are three major repos you are to clone
1.https://github.com/camicroscope, and fork all three repos there, Distro, viewer and Caracal.

Fork it and clone it to your machine

Install docker via https://docs.docker.com/engine/install/, after which you start docker to make sure it's running.

navigate to your machine where you saved the repos and for the Distro repo, try and build with docker-compose -f caMicroscope.yml build and run with docker-compose -f caMicroscope.yml up,

Once everything is up, go to http://localhost:4010/ to see the landing page.

6GB memory is recommended (in total, 2GB RAM + 4GB swap should work well)

The frontend files for the caMicroscope app can be found in the viewer repo, navigate to it and just open the index.html file, it will pop up the landing page in your browser, and you can navigate to other pages on the browser.

You can then go back to the files and make your contributions based on the following guidelines

Welcome to caMicroscope! These contribution tasks are designed to smoothly onboard contributors, showcase skills, and propel the caMicroscope project forward. Tasks are neatly categorized and can be submitted as pull requests or issues. We recommend starting with the Introductory "Getting Started" and "Open Source Communication" categories.

Tasks beyond the introductory stage are scoped to be achievable within a few days to a week for contributors less familiar with caMicroscope. Remember, quality over quantity! A well-executed contribution, coupled with thorough documentation, speaks volumes about your open-source engagement.

No need to be an expert! These tasks are perfect for developing new skills and gaining hands-on experience. Most importantly, enjoy the process. If challenges arise or you need assistance, don't hesitate to start a discussion thread or open an issue. Take your time, experiment, and, if needed, ask for help (we recommend spending about an hour trying to troubleshoot on your own first; issues which take longer than that are good to document/discuss anyway, and will hopefully stop people from becoming frustrated) Thank you for working with us, and happy contributing!

Introductory: Getting Started (Not Directly Pull Request Changes)

Clone caMicroscope and its repositories, especially distro
Install Docker and Docker compose
Run some version of caMicroscope locally (recommended develop.yml)
Add a Slide to your instance of caMicroscope
Experiment with the Viewer Tools
Experiment with additional Tools
Introductory: Open Source Communication (Not Directly Pull Request Changes)

Find or start a relevant discussion for a discussion-like topic
Submit or comment on a relevant issue or pull request
Specifically, make an issue or pull request on something which was unclear, with a proposed fix if applicable
Open Source Support

Reproduce a bug on an open issue and comment about how you’ve done so
UI/UX Design and Programming

Add a dark mode option for the caMicroscope app pages
Report on a small part of caMicroscope’s UI at a task level and some ways it may be improved from a design perspective, including a plan to evaluate these proposed changes
Diagnose and fix the caMicroscope viewer toolbar inconsistently resizing toolbar icons with changing browsers and window sizes
Identify an app or task which is difficult on a mobile device (or other screen size) and propose a low to medium fidelity design solution
Machine Learning and Computer Vision

Try another way to implement the “smart pens” segmentation annotation tool.
Report on a relevant model or computer vision algorithm which could be helpful for caMicroscope’s viewer/annotations
Add a (documented) segmentation ML model to the tfjs_models repository
Add a (documented) classification ML model to the tfjs_models repository
Accessibility

identify one accessibility bug using Web Content Accessibility Guidelines (WCAG) 2.1 standard levels within caMicroscope
Code Performance

Report on a page/app/function which takes a long time for code execution
In addition to the report, you can also propose an improvement (simpler is better!)
Implement caching mechanisms to improve the overall speed and responsiveness of caMicroscope.
Testing

Write tests for some existing caMicroscope functions
After writing tests, set up github actions’ test suite to run these tests automatically
Happy contributions

**Reply by Chy-baby** [reply]
Thanks so much @gbengaoluwadahunsi

**Comment by gbengaoluwadahunsi** [comment]
You are welcome

**Comment by shailpujan88** [comment]
Hi everyone, My name is Shail Pujan from India an Outreachy applicant . So happy to be part of this contribution.
Could you please provide some insights on:

1. Which issues are currently seeking contributions or have entry-level tasks suitable for newcomers like me?
2. Are there any specific coding guidelines or project documentation I should familiarize myself with before diving into contributing code? And any tips regarding setting up the codebase in my local repo
3. Do you recommend any resources or tutorials that could help me get up to speed with the project's technologies or workflows?

Furthermore, I would greatly appreciate any suggestions or tasks you believe would be suitable.

**Reply by We13b-MD** [reply]
Hello shailpujan you are very welcome  here these are steps to help you get started 

Kindly follow the following steps

There are three major repos you are to clone
1.https://github.com/camicroscope, and fork all three repos there, Distro, viewer and Caracal.

Fork it and clone it to your machine

Install docker via https://docs.docker.com/engine/install/, after which you start docker to make sure it's running.

navigate to your machine where you saved the repos and for the Distro repo, try and build with docker-compose -f caMicroscope.yml build and run with docker-compose -f caMicroscope.yml up,

Once everything is up, go to http://localhost:4010/ to see the landing page.

6GB memory is recommended (in total, 2GB RAM + 4GB swap should work well)

The frontend files for the caMicroscope app can be found in the viewer repo, navigate to it and just open the index.html file, it will pop up the landing page in your browser, and you can navigate to other pages on the browser.

You can then go back to the files and make your contributions based on the following guidelines

Welcome to caMicroscope! These contribution tasks are designed to smoothly onboard contributors, showcase skills, and propel the caMicroscope project forward. Tasks are neatly categorized and can be submitted as pull requests or issues. We recommend starting with the Introductory "Getting Started" and "Open Source Communication" categories.

Tasks beyond the introductory stage are scoped to be achievable within a few days to a week for contributors less familiar with caMicroscope. Remember, quality over quantity! A well-executed contribution, coupled with thorough documentation, speaks volumes about your open-source engagement.

No need to be an expert! These tasks are perfect for developing new skills and gaining hands-on experience. Most importantly, enjoy the process. If challenges arise or you need assistance, don't hesitate to start a discussion thread or open an issue. Take your time, experiment, and, if needed, ask for help (we recommend spending about an hour trying to troubleshoot on your own first; issues which take longer than that are good to document/discuss anyway, and will hopefully stop people from becoming frustrated) Thank you for working with us, and happy contributing!

Introductory: Getting Started (Not Directly Pull Request Changes)

Clone caMicroscope and its repositories, especially distro
Install Docker and Docker compose
Run some version of caMicroscope locally (recommended develop.yml)
Add a Slide to your instance of caMicroscope
Experiment with the Viewer Tools
Experiment with additional Tools
Introductory: Open Source Communication (Not Directly Pull Request Changes)

Find or start a relevant discussion for a discussion-like topic
Submit or comment on a relevant issue or pull request
Specifically, make an issue or pull request on something which was unclear, with a proposed fix if applicable
Open Source Support

Reproduce a bug on an open issue and comment about how you’ve done so
UI/UX Design and Programming

Add a dark mode option for the caMicroscope app pages
Report on a small part of caMicroscope’s UI at a task level and some ways it may be improved from a design perspective, including a plan to evaluate these proposed changes
Diagnose and fix the caMicroscope viewer toolbar inconsistently resizing toolbar icons with changing browsers and window sizes
Identify an app or task which is difficult on a mobile device (or other screen size) and propose a low to medium fidelity design solution
Machine Learning and Computer Vision

Try another way to implement the “smart pens” segmentation annotation tool.
Report on a relevant model or computer vision algorithm which could be helpful for caMicroscope’s viewer/annotations
Add a (documented) segmentation ML model to the tfjs_models repository
Add a (documented) classification ML model to the tfjs_models repository
Accessibility

identify one accessibility bug using Web Content Accessibility Guidelines (WCAG) 2.1 standard levels within caMicroscope
Code Performance

Report on a page/app/function which takes a long time for code execution
In addition to the report, you can also propose an improvement (simpler is better!)
Implement caching mechanisms to improve the overall speed and responsiveness of caMicroscope.
Testing

Write tests for some existing caMicroscope functions
After writing tests, set up github actions’ test suite to run these tests automatically
Happy contributions

**Comment by shusonialumnwi** [comment]
Hello community,

I'm Sonia, an Outreachy applicant. After reviewing the projects, caMicroscope has truly caught my attention, and I'm eager to contribute to it. I'm excited about this journey into open source?

Thank you!

**Comment by Africa-Agwu** [comment]
Hi everyone. 
I'm Africa, an Outreachy applicant. I'm so excited to be here . I'm eager to contribute and learn from this project.

**Comment by Ayesh-07** [comment]
Hello @gbengaoluwadahunsi  @birm ,

I hope this message finds you well. I wanted to reach out to express my interest in contributing to the caMicroscope project. As an Outreachy applicant, I am eager to contribute to open-source projects, and I noticed that there is an open issue titled "Reload button margin needed in slides page (table.html) https://github.com/camicroscope/caMicroscope/issues/737" that I believe I can help with.

I have already fixed the issue in my local repository and would like to contribute this fix to the project. Could you please guide me on the next steps to create a pull request and get my changes reviewed and merged?

**Reply by gbengaoluwadahunsi** [reply]
hi @Ayesh-07 , just seeing this, I hope you were able to make your PR. happy contribution.

**Comment by kanonkartik** [comment]
Hello @birm  @gbengaoluwadahunsi , 
My name is Kartik Kakodiya , and I am excited to express my interest in contributing to the caMicroscope project as part of the Outreachy program.
After carefully reviewing the project description and understanding its goals, I believe I can make meaningful contributions to creating a standalone executable for caMicroscope. My background in Backend web development , Frontend Web Development such as software development, web hosting,  has equipped me with the necessary expertise to tackle the challenges outlined in the project scope.

**Comment by gbengaoluwadahunsi** [comment]
Hello Kartik,

You are welcome to this awesome project, with the beauty ML research is
adding  to biomedical research, we are  thrilled to be contributors to this.

If you need any information please let me know.

Happy  contributions.

**Comment by arghya-align-pix-design** [comment]
Hello everyone.I am Arghya, I am an outreachy Applicant, i came across caMicroscope in the project section and i found my skills aligned with the project enlisted there, i am a full stack developer , i have been going through the code bases and been studying how work is being done here. I am very excited to learn and be able to contribute here and be a part of this community. Thank you.

**Comment by gbengaoluwadahunsi** [comment]
Welcome aboard!

**Comment by Rishi2003Das** [comment]
Hello everyone! I am **Rishi Das** from India. I am willing to participate as a **contributor in GSOC 2025** and while checking out the past projects I find your organization very interesting and relatable to my work. I am an undergrad engineering student studying in my 3rd year of graduation. I am very enthusiastic to work in fields of AI/ML and I am also doing **research in this field of medical imaging**. I mainly work in Computer Vision projects related to **Medical Imaging like Cancer detection and classification**. I have participated in a few competitions and hackathons too as an ML Engineer and did several challenges in Kaggle. It's my passion to build in new model architectures and train it in large datasets. Hope you like my work and introduction. I am interested to contribute in your projects so, please mention what are the projects that you are aiming for in future to work for GSOC '25. So, that I can contribute to your project. 
Thanking you,
Rishi Das.
[LinkedIn](https://www.linkedin.com/in/rishi-das-522537250/)
[Kaggle](https://www.kaggle.com/rishi2003das)
[Email](rishikakalidas@gmail.com)
